coffee-chats/src/main/webapp/src/util/fetch.ts

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+/**
+ * A hook that fetches JSON data from a URL using a GET request
+ *
+ * @param url: URL to fetch data from
+ */
+export function useFetch(url: string): any {
+  const [data, setData] = React.useState(null);
+
+  React.useEffect(() => {
+    (async () => {
+      const response = await fetch(url);
+      const json = await response.json();
+
+      if (!response.ok && json.loginUrl) {
+        window.location.href = json.loginUrl;
+      } else {
+        setData(json);
+      }
+    })();
+  }, [url]);
```

Comment:
I don't think this is the correct use of useEffect, the url should never change within a function call. Perhaps the calling component may use useEffect and the url as a dependency in this way, and call useFetch within the hook there.

Reply:
`useFetch` is already a hook and it already calls useEffect with correct dependencies, I don't see a problem here. What you're suggesting will lead to `useEffect` being called within the `useEffect` callback (if I understood you correctly).